src/FlatSharp.Compiler/Visitors/FieldVisitor.cs

```diff
@@ -102,7 +77,69 @@ public override FieldDefinition VisitField_decl([NotNull] FlatBuffersParser.Fiel
                 }
             });
 
-            return this.definition;
+            return definition;
+        }
+
+        private void ParseIdMetadata(IDictionary<string, string> metadata)
+        {
+            if (!metadata.TryParseIntegerMetadata("id", out var index))
+            {
+                if (index == MetadataHelpers.DefaultValueIfPresent)
```

Comment:
Not sure I like the naming here. Or -- I don't like the "defaultvalueifpresent" being on the MetaDataHelpers class. It has a specific meaning in this context, so I think we should define it in this class.

Alternatively, I think the semantic you are really trying to capture here is nullable. Why not have TryParseIntegerMetadata return a nullable int?

Reply:
Also, I wonder how useful this validation is. The compiler is supposed to be at least somewhat dumb. The FlatSharp library (ie, the thing that actually generates the interesting code) will throw a fit if there are two properties with the same index, so we could just let people define whatever values they want here.